### PR TITLE
planet check for ships, rather than system

### PR DESCRIPTION
**Pull request by flowers9**
**Refactor:**

## Fix Details
When checking to see if the outfitter can be accessed, see if non-disabled ships are present on the planet, rather than in the system, as only the former will always be available for outfitting.

## Testing Done
It's a pretty trivial change, and the ShopPanel::DrawShipsSidebar() code already does this check and won't display ships not on planet (doing the check in ShopPanel::CanShowInSideBar()).  So I just tried it out on a new game a saved game and it worked as expected.

### CODE REVIEW

**Review by quyykk — APPROVED** [review]

**Review by TomGoodIdea — APPROVED** [review]

**Review by lumbar527 — APPROVED** [review]

**Review by RisingLeaf — APPROVED** [review]

**Review by Hecter94 — APPROVED** [review]

**Review by Quantumshark — APPROVED** [review]

**Review by Hurleveur — APPROVED** [review]

**Review by Hurleveur — COMMENTED** [review]
We dont have many cases of multiple shipyards in a system yet (Sol), and this applies to someone parking ships on other worlds than the shipyard one.
The case where this would be detrimental is if you cant unpark the ship.
Actually, thinking about it more, parking a ship is not done on a planet but in a system, and all ships presumably follow you, except if they cant. We probably should cover the case where they cant some other way if we change it to this and adapt the playerinfodisplay to show what *outfitter* the ship is on if there is one in the system.

I think I got carried away by the many reviews x)

**Comment by quyykk** [comment]
>  The case where this would be detrimental is if you cant unpark the ship.

You can can always unpark a ship?

>  parking a ship is not done on a planet but in a system

you can only park a ship on a planet though?

> adapt the playerinfodisplay to show what outfitter the ship is on if there is one in the system.

yeah, but that's already a problem without this PR so

File: source/PlanetPanel.cpp [hunk 1]
```diff
@@ -127,7 +127,7 @@ void PlanetPanel::Draw()
 
 		if(planet.HasOutfitter())
 			for(const auto &it : player.Ships())
-				if(it->GetSystem() == &system && !it->IsDisabled())
+				if(it->GetPlanet() == &planet && !it->IsDisabled())
```
Comment (TomGoodIdea) [inline source/PlanetPanel.cpp:130]:
```suggestion
				if(it->GetPlanet() == &planet)
```

File: source/PlanetPanel.cpp [hunk 2]
```diff
@@ -189,7 +189,7 @@ bool PlanetPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, b
 	else if(key == 'o' && hasAccess && planet.HasOutfitter())
 	{
 		for(const auto &it : player.Ships())
-			if(it->GetSystem() == &system && !it->IsDisabled())
+			if(it->GetPlanet() == &planet && !it->IsDisabled())
```
Comment (TomGoodIdea) [inline source/PlanetPanel.cpp:192]:
```suggestion
			if(it->GetPlanet() == &planet)
```

**Review by TomGoodIdea — COMMENTED** [review]
You can actually remove `IsDisabled()` checks.

**Comment by Hurleveur** [comment]
You can park a ship in a system with no planets!
Also you cant unpark one if it doesnt pass the flight checks
Also no that third one isnt a trouble without this PR since you dont need to know which planet it is on.

**Comment by flowers9** [comment]
The relevent bit is that in ShopPanel::DrawShipsSidebar(), it will only allow the player to outfit ships that pass the CanShowInSidebar() check, which simply checks if the ship is on the planet.  In other words, this PR is just making sure that the Outfitter will only show up if the player actually has ships that can be outfitted.  While there may be edge cases to consider (with parking. disabling, or landing on multiple planets), this PR is not effected by them, and making this consistent with ShopPanel::CanShowInSidebar() means it'll be easier to deal with them (by making the checks identical in both places) if anyone finds one.

**Comment by Hurleveur** [comment]
Maybe it should be changed in the other way?
Or would that enable modifying ships disabled in the system? (We can account for that)
That said if thats true then not many ppl got into the edge case to complain about it

**Comment by warp-core** [comment]
#9032 supercedes this, and that's merged now.

**Pull request closed without merging**